File: 1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.py
class Solution(object):
    def minKBitFlips(self, nums, k):
        n = len(nums)
        x = 0
        y = 0
        z = [0] * n

        for i in range(n):
            if i >= k:
                y ^= z[i - k]
            
            if nums[i] == y:
                if i + k > n:
                    return -1
                x += 1
                y ^= 1
                if i < n:
                    z[i] = 1
        
        return x

File: 1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.java
class Solution {
    public int minKBitFlips(int[] nums, int k) {
        int n = nums.length;
        int[] z = new int[n];
        int x = 0;
        int y = 0;

        for (int i = 0; i < n; i++) {
            if (i >= k) {
                y ^= z[i - k];
            }

            if (nums[i] == y) {
                if (i + k > n) {
                    return -1;
                }
                x++;
                y ^= 1;
                if (i < n) {
                    z[i] = 1;
                }
            }
        }

        return x;
    }
}

